blog/experimental-support-for-gophers-in-wasmCloud/index.mdx

```diff
@@ -0,0 +1,66 @@
+---
+title: "Experimental Support for Gophers in wasmCloud"
+image: "./blobby-versus-globby.jpg"
+date: 2023-11-21T9:00:00-04:00
+author: "Brooks Townsend"
+author_profile: "https://www.linkedin.com/in/brooks-townsend/"
+description: "wasmCloud now comes with experimental component support for Go"
+categories: ["Golang", "wasmcloud"]
+draft: false
+---
+
+import ReactPlayer from 'react-player'
+
+![from wasmcloud specific dependencies to zero dependencies](./blobby-versus-globby.jpg)
+
+wasmCloud now comes with experimental support for Go, via TinyGo compiler, taking full advantage of the WebAssembly component model. This is an exciting moment as, for the first time, Go developers can build cloud native Wasm applications with components and run them in wasmCloud in an open, dependency-free and standards-based way.
+
+<!--truncate-->
+
+:::info
+
+*[TinyGo](https://tinygo.org/)* is *"a Go compiler for small places"*. It is a language designed specifically to work on embedded systems and WebAssembly. For Gophers, TinyGo is commonly a simple change from running `go build` to running `tinygo build`, as TinyGo [supports many of the Go standard library packages](https://tinygo.org/docs/reference/lang-support/stdlib/).
+
+:::
+
+Some of the best-known organizations develop in Go; Google, Uber, Netflix, Twitch to name a few. There's a growing appetite for Wasm amongst Go developers in all industries and so, with Go support now in wasmCloud, they have a community-driven, open environment to get WebAssembly projects off the ground.
+
+### Rebuilding Actors as Components
+
+wasmCloud has supported Go for [some time](https://wasmcloud.com/blog/example_creating_webassembly_actor_in_go_with_tinygo). As many in the community know, there are several TinyGo example Wasm applications, including `echo actor`, `blobby actor` and `kv-counter actor`, which were written around WASI-Preview 1 specifications. These examples compile to wasm modules and use our stable ABI and, as such, they rely on a few wasmCloud-specific dependencies to successfully run there.
+
+As wasmCloud is engineered around WASI-Preview 2, we have rewritten these actors to use WASI contracts; completely removing any dependencies. Smithy is a thing of the past as we focus on [WIT](https://cosmonic.com/blog/engineering/wit-cheat-sheet) to bind together polyglot components. Take a look at our new WASIfied apps `[echo-http-server](https://github.com/wasmCloud/wasmCloud/tree/main/examples/golang/actors/http-echo-tinygo)` and `[globby-wasi](https://github.com/jordan-rash/globby-wasi)` (blobby reimagined in Go!).
```

Comment:
Hmm I think the backticks here are messing with the markdown link

Reply:
It's the html `<!-- truncate -->` comment breaking the MDX formatting because technically it should be a `{/* JSX Comment */}`. This is [resolved in Docusaurus 3](https://docusaurus.io/docs/migration/v3#comments-option) and I'll get to that upgrade tomorrow.

Reply:
Nevermind, I see the issue.

```suggestion
As wasmCloud is engineered around WASI-Preview 2, we have rewritten these actors to use WASI contracts; completely removing any dependencies. Smithy is a thing of the past as we focus on [WIT](https://cosmonic.com/blog/engineering/wit-cheat-sheet) to bind together polyglot components. Take a look at our new WASIfied apps [echo-http-server](https://github.com/wasmCloud/wasmCloud/tree/main/examples/golang/actors/http-echo-tinygo) and [globby-wasi](https://github.com/jordan-rash/globby-wasi) (blobby reimagined in Go!).
```